R/learner_xgboost_surv_xgboost.R

```diff
@@ -71,7 +69,6 @@ LearnerSurvXgboost = R6Class("LearnerSurvXgboost",
         normalize_type              = p_fct(c("tree", "forest"), default = "tree", tags = "train"),
         nrounds                     = p_int(1L, tags = "train"),
         nthread                     = p_int(1L, default = 1L, tags = c("train", "threads")),
-        ntreelimit                  = p_int(1L, tags = "predict"),
```

Comment:
why was this removed?

Reply:
So in the `predict.XGBoost` method I saw that `iterationrange` is used instead (which we include either way): 
![ntreelimit](https://github.com/mlr-org/mlr3extralearners/assets/10909555/3db1cdcd-4a20-4473-a6d7-837e3964f70a)